x-pack/plugins/data_enhanced/public/search/sessions_mgmt/components/status.tsx

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiIcon, EuiLoadingSpinner, EuiToolTip } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { ReactElement } from 'react';
+import { STATUS, UISession } from '../../../../common/search/sessions_mgmt';
+import { dateString } from '../lib/date_string';
+import { StatusDef as StatusAttributes, TableText } from './';
+
+// Shared helper function
+export const getStatusText = (statusType: string): string => {
+  switch (statusType) {
+    case STATUS.IN_PROGRESS:
+      return i18n.translate('xpack.data.mgmt.searchSessions.status.label.inProgress', {
+        defaultMessage: 'In progress',
+      });
+    case STATUS.EXPIRED:
+      return i18n.translate('xpack.data.mgmt.searchSessions.status.label.expired', {
+        defaultMessage: 'Expired',
+      });
+    case STATUS.CANCELLED:
+      return i18n.translate('xpack.data.mgmt.searchSessions.status.label.cancelled', {
+        defaultMessage: 'Cancelled',
+      });
+    case STATUS.COMPLETE:
+      return i18n.translate('xpack.data.mgmt.searchSessions.status.label.complete', {
+        defaultMessage: 'Complete',
+      });
+    case STATUS.ERROR:
+      return i18n.translate('xpack.data.mgmt.searchSessions.status.label.error', {
+        defaultMessage: 'Error',
+      });
+    default:
+      // eslint-disable-next-line no-console
+      console.error(`Unknown status ${statusType}`);
+      return statusType;
+  }
+};
+
+interface StatusIndicatorProps {
+  now?: string;
+  session: UISession;
+  timezone: string;
+}
+
+// Get the fields needed to show each status type
+// can throw errors around date conversions
+const getStatusAttributes = ({
+  now,
+  session,
+  timezone,
+}: StatusIndicatorProps): StatusAttributes | null => {
+  let expireDate: string;
+  if (session.expires) {
+    expireDate = dateString(session.expires!, timezone);
+  } else {
+    expireDate = i18n.translate('xpack.data.mgmt.searchSessions.status.expireDateUnknown', {
+      defaultMessage: 'unknown',
+    });
+  }
+
+  switch (session.status) {
+    case STATUS.IN_PROGRESS:
+      try {
+        return {
+          textColor: 'default',
+          icon: <EuiLoadingSpinner />,
+          label: <TableText>{getStatusText(session.status)}</TableText>,
+          toolTipContent: i18n.translate(
+            'xpack.data.mgmt.searchSessions.status.message.createdOn',
+            {
+              defaultMessage: 'Expires on {expireDate}',
```

Comment:
I'm not sure I understand why we show the expiration here in the tooltip but we don't show it in the actual table for in-progress.